chainer/functions/array/reshape.py

```diff
@@ -51,12 +51,36 @@ def reshape(x, shape):
     """Reshapes an input variable without copy.
 
     Args:
-        x (~chainer.Variable): Input variable.
-        shape (tuple of ints): Target shape.
+        x (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`): Input variable.
+        shape (:class:`tuple` of :class:`int` s):
+            The **size** of shape (**size** means the number of elements) must
```

Comment:
Add an brief explanation of this argument before its constraint (e.g. "Expected shape of the output array").